src/pigpio.cc

```diff
@@ -707,6 +707,199 @@ NAN_METHOD(gpioNotifyClose) {
   }
 }
 
+/* ------------------------------------------------------------------------ */
+/* Waves                                                                    */
+/* ------------------------------------------------------------------------ */
+
+NAN_METHOD(gpioWaveClear) {
+  int rc = gpioWaveClear();
+  if (rc < 0) {
+    return ThrowPigpioError(rc, "gpioWaveClear");
+  }
+}
+
+NAN_METHOD(gpioWaveAddNew) {
+  int rc = gpioWaveAddNew();
+  if (rc < 0) {
+    return ThrowPigpioError(rc, "gpioWaveAddNew");
+  }
+}
+
+NAN_METHOD(gpioWaveAddGeneric) {
+  if (info.Length() < 1 || !info[0]->IsArray()) {
+    return Nan::ThrowError(Nan::ErrnoException(EINVAL, "gpioWaveAddGeneric", ""));
+  }
+
+  
+  v8::Local<v8::Array> array = info[0].As<v8::Array>();
+  unsigned numPulses = array->Length();
+
+  gpioPulse_t pulses[numPulses];
+
+  for (unsigned i = 0; i < array->Length(); i++) {
+    if (Nan::Has(array, i).FromJust()) {
+      
+      v8::Local<v8::Value> element = array->Get(i);
+      
+      if (!element->IsObject()) {
+        return Nan::ThrowError(Nan::ErrnoException(EINVAL, "gpioWaveAddGeneric", "")); 
+      }
+
+      v8::Local<v8::Object> pulse = v8::Local<v8::Object>::Cast(element);
+
+      v8::Isolate * isolate = v8::Isolate::GetCurrent();
```

Comment:
`       v8::Isolate * isolate = v8::Isolate::GetCurrent();` can be removed as the code that is currently using `isolate` will be modified and `isolate` will no longer be needed.